users/token/token.go

```diff
@@ -0,0 +1,154 @@
+		t.hmacSampleSecret = []byte(mainflux.Env(envTokenSecret, defTokenSecret))
+		t.tokenDuration, _ = strconv.Atoi(mainflux.Env(envTokenDuration, defTokenDuration))
+		t.url = mainflux.Env(envTokenResetEndpoint, defTokenResetEndpoint)
+		logLevel := mainflux.Env(envTokenLogLevel, defTokenLogLevel)
+		l, err := logger.New(os.Stdout, logLevel)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		t.logger = l
+	})
+	return t
+}
+
+// Generate generate new random token with defined TTL.
+// offset can be used to manipulate token validity in time
+// useful for testing.
+func Generate(email string, offset int) (string, error) {
+	return instance().generate(email, offset)
+}
+
+// Verify verifies token validity
+func Verify(tok string) (string, error) {
+	return instance().verify(tok)
+}
+
+// SendToken sends password recovery link to user
+func SendToken(host, email, token string) {
+	instance().sendToken(host, email, token)
+}
+
+func (t *tokenizer) generate(email string, offset int) (string, error) {
+
+	exp := t.tokenDuration + offset
+	if exp < 0 {
+		exp = 0
+	}
+	expires := time.Now().Add(time.Minute * time.Duration(exp))
+	nbf := time.Now()
+
+	// Create a new token object, specifying signing method and the claims
+	// you would like it to contain
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"email": email,
+		"exp":   expires.Unix(),
+		"nbf":   nbf.Unix(),
+	})
+
+	// Sign and get the complete encoded token as a string using the secret
+	tokenString, err := token.SignedString(t.hmacSampleSecret)
+	fmt.Println(tokenString)
+	return tokenString, err
+}
+
+// Verify verifies token validity
+func (t *tokenizer) verify(tok string) (string, error) {
+	email := ""
+	token, err := jwt.Parse(tok, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			t.logger.Error(ErrWrongSignature.Error())
+			return nil, ErrWrongSignature
+		}
+
+		// hmacSampleSecret is a []byte containing your secret, e.g. []byte("my_secret_key")
+		return t.hmacSampleSecret, nil
+	})
+	if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
+		if claims.VerifyExpiresAt(time.Now().Unix(), false) == false {
+			t.logger.Error(ErrExpiredToken.Error())
+			return "", ErrExpiredToken
+		}
+		email = claims["email"].(string)
+
+	} else {
+		return email, err
+	}
+	return email, nil
+}
+
+// SendToken sends password recovery link to user
+func (t *tokenizer) sendToken(host, email, token string) {
+	body := t.buildBody(host, email, token)
+	mail.Send([]string{email}, body)
+}
+
+// Builds recovery email body
+func (t *tokenizer) buildBody(host, email, token string) []byte {
+	msg := []byte(fmt.Sprintf("To: %s\r\n"+
+		"Subject: Reset Password!\r\n"+
+		"\r\n"+
+		"You have initiated password reset.\r\n"+
+		"Follow the link below to reset password.\r\n"+
+		"%s%s?token=%s", email, host, t.url, token))
+
+	return msg
+}
```

Comment:
Here you shouldn't have anything related to the mail or password reset. Only token logic.